### PR TITLE
feat(osm/install): Decouple releaseNamespace and osmNamespace

### PR DESCRIPTION
**Pull request by ksubrmnn**
**Description**:
This PR allows the release-namespace for the osm installation to be different than the osm namespace.

By creating an osmNamespace field in the helm Values file, users can hardcode the namespace where the osm controller will be deployed. This way any OSM operators can allow users to specify where their helm release will live while still installing osm in a predictable namespace. 

**Affected area**:

- New Functionality      [ ]
- Documentation          [ ]
- Install                [X]
- Control Plane          [ ]
- CLI Tool               [ ]
- Certificate Management [ ]
- Networking             [ ]
- Metrics                [ ]
- SMI Policy             [ ]
- Security               [ ]
- Tests                  [ ]
- CI System              [ ]
- Performance            [ ]
- Other                  [ ]

Please answer the following questions with yes/no.

- Does this change contain code from or inspired by another project? If so, did you notify the maintainers and provide attribution?
No

### CODE REVIEW

File: charts/osm/templates/_helpers.tpl [hunk 1]
```diff
@@ -1,63 +1,4 @@
-{{/* vim: set filetype=mustache: */}}
-{{/*
-Expand the name of the chart.
```
Comment (ksubrmnn) [inline charts/osm/templates/_helpers.tpl:3]: I didn't see these used anymore, please let me know if I'm mistaken!

File: charts/osm/values.yaml [hunk 2]
```diff
@@ -55,6 +55,7 @@ OpenServiceMesh:
   controllerLogLevel: trace
   enforceSingleMesh: false
   webhookConfigNamePrefix: osm-webhook
+  osmNamespace: ""
```
Comment (ritazh) [inline charts/osm/values.yaml:58]: Can we add a comment here and in the chart readme that if `osmNamespace` is not provided, then the release namespace is used as the namespace for where the osm controller will be installed?

**Review by SanyaKochhar — APPROVED** [review]

**Review by nshankar13 — APPROVED** [review]

**Comment by shashankram** [comment]
@ksubrmnn are we good with https://github.com/openservicemesh/osm/issues/2141#issuecomment-740742818?

File: charts/osm/values.schema.json [hunk 3]
```diff
@@ -397,6 +397,15 @@
                     "examples": [
                         true
                     ]
+                },
+                "osmNamespace": {
+                    "$id": "#/properties/OpenServiceMesh/properties/osmNamespace",
+                    "type": "string",
+                    "title": "The namespace of the osm controller",
+                    "description": "Indicates the namespace in which the osm controller will be installed.",
```
Comment (michelleN) [inline charts/osm/values.schema.json:405]: nit:
title: "The namespace of the osm control plane components"
description: "Indicates the namespace in which the osm control plane components will be installed

Comment (ksubrmnn) [reply charts/osm/values.schema.json:405]: I'll update this nit separately so I don't dismiss stale reviews.

**Review by ritazh — APPROVED** [review]

**Pull request merged**